pkg/builder/options.go

```diff
@@ -58,17 +58,17 @@ type Predicates struct {
 
 // ApplyToFor applies this configuration to the given ForInput options.
 func (w Predicates) ApplyToFor(opts *ForInput) {
-	opts.predicates = w.predicates
+	opts.predicates = append(opts.predicates, w.predicates...)
```

Comment:
This is an api breach. It is debatable if the previous approach or the current one is "correct" but since we have the previous one already, I think it makes sense to stick with it.

This approach also has the drawback that you can not reset the `predicates` via `opts`.

Reply:
What we probably should do instead is just nilcheck the predicates before setting them

Reply:
> This is an api breach. It is debatable if the previous approach or the current one is "correct" but since we have the previous one already, I think it makes sense to stick with it.

Well, having a variable number of Option parameters for the `For()` method definitely implies that each of those Option things will be "applied", no? It's a surprising user experience to have all but the last supplied Option do nothing at all.

> This approach also has the drawback that you can not reset the `predicates` via `opts`.

There should probably be a `SetOptions` method then for that?

Reply:
> Well, having a variable number of Option parameters for the For() method definitely implies that each of those Option things will be "applied", no? It's a surprising user experience to have all but the last supplied Option do nothing at all.

Yeah unless you explicitly set it to an empty list. Nilchecking solves the problem you have while still allowing to explicitly reset them by using a non-nil but empty `[]Predicate`

Reply:
If we want to proceed with this change, we can include it in the v0.7.x breaking release